### PR TITLE
feat(wdio): refactored enum, added wdio ChainablePromise support

### PR DESCRIPTION
**Pull request by vitalyiegorov**

### CODE REVIEW

File: packages/wdio/src/command/test-id-index.command.ts [hunk 1]
```diff
@@ -9,5 +10,6 @@ export const testID$$Index: ElsIndexSelectorFn = async (testID, index, context =
         throw new Error(`Cannot get item by testID "${testID}" with index "${index}"`);
     }
 
-    return elements[index];
+    // eslint-disable-next-line @typescript-eslint/return-await
+    return elements[index] as unknown as ChainablePromiseElement<WebdriverIO.Element>;
```
Comment (habiiev) [inline packages/wdio/src/command/test-id-index.command.ts:14]: Why didn't you want to return await here? It wouldn't break your logic

Comment (habiiev) [inline packages/wdio/src/command/test-id-index.command.ts:14]: Wouldn't it be better to change `testID$$` type to return `ChainablePromiseElement`

File: packages/wdio/src/component/component/component.ts [hunk 2]
```diff
@@ -65,15 +66,15 @@ export class Component {
         return await (await this.getChildEl(selector)).getSize();
     }
 
-    async getChildEl(selector: string): Promise<WebdriverIO.Element> {
-        return await this.elSelectorFn(selector);
+    getChildEl(selector: string): ChainablePromiseElement<WebdriverIO.Element> {
+        return this.elSelectorFn(selector) as ChainablePromiseElement<WebdriverIO.Element>;
```
Comment (habiiev) [inline packages/wdio/src/component/component/component.ts:70]: Why not change the type of `elSelectorFn` to return `ChainablePromiseElement`

File: packages/wdio/src/component/component/get-component/get-component.ts [hunk 3]
```diff
@@ -2,8 +2,10 @@ import { defaultComponentConfig } from '../../default-component.config';
 import { createComponentWithSelectorsProxy } from '../../util';
 import { Component } from '../component';
 
-import type { Enum } from '../../../type';
 import type { ComponentConfigInterface, ComponentWithSelectorsCtor } from '../../type';
+import type { Enum } from '@rnw-community/shared';
+
+export type Type<T extends Component> = new (config: ComponentConfigInterface) => T;
```
Comment (habiiev) [inline packages/wdio/src/component/component/get-component/get-component.ts:8]: I don't think name `Type` is the best choice, it's too generic

Wouldn't `ComponentType` be better?

**Review by habiiev — CHANGES_REQUESTED** [review]

File: packages/wdio/src/component/component/component.ts (hunk 2 above)
Comment (vitalyiegorov) [reply packages/wdio/src/component/component/component.ts:70]: Ohhh, this brings huge amount of unfixable TS issues

**Review by habiiev — DISMISSED** [review]

File: packages/wdio/src/component/component/get-component/get-component.spec.ts [hunk 4]
```diff
@@ -9,168 +8,16 @@ enum SelectorsEnum {
     Button = 'Selectors.Button',
 }
 
-class Component extends getComponent(SelectorsEnum) {}
+class CustomComponent extends getComponent(SelectorsEnum) {}
 
-// eslint-disable-next-line max-lines-per-function,max-statements
 describe('getComponent', () => {
-    it('should throw error on calling not supported proxy method', () => {
```
Comment (habiiev) [inline packages/wdio/src/component/component/get-component/get-component.spec.ts:16]: why did you remove all of these tests?

**Review by habiiev — APPROVED** [review]
